testing.rb

```diff
@@ -0,0 +1,313 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    self
+  end
```

Comment:
@javitocor please follow the feedback from @cvilla714  here.

When your `my_each` receives a hash and a proc in it's input, it should behave like this:

```
        enum = { a: 1, b: 2, c: 3, d: 4, e: 5 }
        my_each_output = ''
        block = proc { |num| my_each_output += num.to_s }
        enum.each(&block)
        each_output = my_each_output.dup
        my_each_output = ''
        enum.my_each(&block)

# These two variables should be the same.
        my_each_output == each_output
```

Please check that this tests passes in your computer locally before re-submitting for review.